elftools/dwarf/compileunit.py

```diff
@@ -73,14 +76,84 @@ def get_top_DIE(self):
         """ Get the top DIE (which is either a DW_TAG_compile_unit or
             DW_TAG_partial_unit) of this CU
         """
-        return self._get_DIE(0)
+        dm = self._diemap
+
+        # Note that a top DIE always has minimal offset, no bisect required.
+        if dm and dm[0] == self.cu_die_offset:
+            return self._dielist[0]
+        top = DIE(
+                cu=self,
+                stream=self.dwarfinfo.debug_info_sec.stream,
+                offset = self.cu_die_offset)
+
+        self._dielist.insert(0, top)
+        dm.insert(0, self.cu_die_offset)
+
+        return top
 
     def iter_DIEs(self):
         """ Iterate over all the DIEs in the CU, in order of their appearance.
             Note that null DIEs will also be returned.
         """
-        self._parse_DIEs()
-        return iter(self._dielist)
+        return self._iter_DIE_subtree(self.get_top_DIE())
+
+    def iter_DIE_children(self, die):
+        """ Given a DIE, yields either its children, without null DIE list
+            terminator, or nothing, if that DIE have no children.
+
+            The null DIE terminator is saved in that DIE when iteration ended.
+        """
+        if not die.has_children:
+            return
+
+        dm = self._diemap
+        dl = self._dielist
+        s = die.stream
```

Comment:
Don't use these abbreviations, please

Reply:
Accessing attributes in [slower][1] than locals. It's significant in a hot code. However, I can use full variable names.

[1]: https://stackoverflow.com/questions/28597014/python-why-is-accessing-instance-attribute-is-slower-than-local

Reply:
Yep, I know that. Does it actually affect the profile of DWARF parsing with pyelftools? I find that extremely unlikely given all the other things this code is doing